File: app/src/main/java/com/fuchsia/shotokanwkf/Admob.java
package com.fuchsia.shotokanwkf;

import android.content.Context;
import androidx.annotation.NonNull;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.LoadAdError;
import com.google.android.gms.ads.MobileAds;
import com.google.android.gms.ads.initialization.InitializationStatus;
import com.google.android.gms.ads.initialization.OnInitializationCompleteListener;
import com.google.android.gms.ads.interstitial.InterstitialAd;
import com.google.android.gms.ads.interstitial.InterstitialAdLoadCallback;


public class Admob {

   public  static InterstitialAd mInterstitialAd;

   public static  void loadInter(Context context){

       MobileAds.initialize(context, new OnInitializationCompleteListener() {
           @Override
           public void onInitializationComplete(InitializationStatus initializationStatus) {}
       });

       AdRequest adRequest = new AdRequest.Builder().build();

       InterstitialAd.load(context,context.getString(R.string.interstitialId), adRequest,
               new InterstitialAdLoadCallback() {
                   @Override
                   public void onAdLoaded(@NonNull InterstitialAd interstitialAd) {

                       mInterstitialAd = interstitialAd;
                   }

                   @Override
                   public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {

                       mInterstitialAd = null;
                   }

               });

   }
}

File: app/src/main/java/com/fuchsia/shotokanwkf/Admob.kt
package com.fuchsia.shotokanwkf

import android.content.Context
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.LoadAdError
import com.google.android.gms.ads.MobileAds
import com.google.android.gms.ads.interstitial.InterstitialAd
import com.google.android.gms.ads.interstitial.InterstitialAdLoadCallback

object Admob {
    var mInterstitialAd: InterstitialAd? = null
    fun loadInter(context: Context) {
        MobileAds.initialize(context) { }
        val adRequest = AdRequest.Builder().build()
        InterstitialAd.load(context, context.getString(R.string.interstitialId), adRequest,
            object : InterstitialAdLoadCallback() {
                override fun onAdLoaded(interstitialAd: InterstitialAd) {
                    mInterstitialAd = interstitialAd
                }

                override fun onAdFailedToLoad(loadAdError: LoadAdError) {
                    mInterstitialAd = null
                }
            })
    }
}